src/main/java/entity/Portfolio.java

```diff
@@ -43,33 +43,33 @@ public double getTotalValue() {
      * @return an Optional containing the UserStock if found, or an empty Optional otherwise
      */
     public Optional<UserStock> getUserStock(String ticker) {
-        return Optional.ofNullable(stocks.get(ticker));
+        return Optional.ofNullable(userStocks.get(ticker));
     }
 
     /**
      * Retrieve all UserStock objects in the portfolio.
      *
      * @return a collection of all UserStock objects
      */
-    public Collection<UserStock> getAllStocks() {
-        return stocks.values();
+    public Collection<UserStock> getAllUserStocks() {
+        return userStocks.values();
     }
 
     /**
      * Adds or updates a stock in the portfolio.
      *
      * @param userStock the UserStock to add or update
      */
-    public void addStock(UserStock userStock) {
-        stocks.put(userStock.getStock().getTicker(), userStock);
+    public void addUserStock(UserStock userStock) {
+        userStocks.put(userStock.getStock().getTicker(), userStock);
     }
 
     /**
      * Removes a stock from the portfolio.
      *
      * @param userStock the UserStock to remove
      */
```

Comment:
Add a new test: If it starts with an initial position of -200, and you `buy` 200 shares. The stock is supposed to be removed from the `portfolio`, however, the current implementation will not remove the `userStock`. Update the entity or the interactors of `execute_buy` / `execute_sell`.